core/src/main/kotlin/com/rudderstack/sdk/kotlin/core/internals/policies/backoff/ExponentialBackOffPolicy.kt

```diff
@@ -0,0 +1,55 @@
+package com.rudderstack.sdk.kotlin.core.internals.policies.backoff
+
+import com.rudderstack.sdk.kotlin.core.internals.utils.InternalRudderApi
+import java.security.SecureRandom
+import kotlin.math.pow
+
+internal const val MIN_INTERVAL = 10L
+internal const val MAX_INTERVAL = 60_000L
+internal const val DEFAULT_INTERVAL = 3000L
+
+internal const val MIN_BASE = 1.1
+internal const val MAX_BASE = 5.0
+internal const val DEFAULT_BASE = 2.0
```

Comment:
Do the following changes:

1. Convert all the value to seconds and avoid decimal values. We can achieve the same behaviour by having the values as second and this will make our code more readable, maintainable and simpler.
2. Let's set MIN_INTERVAL to 3 seconds; that is the default value.
3. Rename the `..._INTERVAL` to `..._DELAY_IN_SEC` or `..._INITIAL_DELAY_IN_SEC`.
4. Set the `MIN_BASE` to `2`.
5. Rename the constants to signify if they are in seconds or milliseconds. The variable name should give a clear idea of the unit.

```suggestion
internal const val MIN_DELAY_IN_SEC = 3
internal const val MAX_DELAY_IN_SEC = 6
internal const val DEFAULT_DELAY_IN_SEC = 3

internal const val MIN_BASE_IN_SEC = 2
internal const val MAX_BASE_IN_SEC = 5
internal const val DEFAULT_BASE_IN_SEC = 2
```

Reply:
Everthing else except `BASE` is fine. Base is a unitless value.